### PR TITLE
Panic hook

### PR DESCRIPTION
**Pull request by arkpar**
Stops the process if any thread panics.

### CODE REVIEW

**Comment by tomaka** [comment]
I don't see where this aborts the process.

But instead of doing `process::abort()`, we could add this to the Cargo.toml:

```toml
[profile.debug]
panic = "abort"

[profile.release]
panic = "abort"
```

As this also enables lots of optimizations that aren't possible otherwise.

**Comment by tomaka** [comment]
The big downside of `panic=abort` is that running tests still requires unwinding, and therefore the whole dependencies tree needs to be compiled separately for `cargo build` and `cargo test`.

**Comment by arkpar** [comment]
@tomaka Thanks for pointing this out. For some reason I was sure we already had `panic="abort"` in Carto.toml

File: polkadot/cli/src/panic_hook.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+static ABOUT_PANIC: &str = "
+This is a bug. Please report it at:
+
+    https://github.com/paritytech/polkadot/issues/new
+";
+
+fn panic_hook(info: &PanicInfo) {
+	let location = info.location();
```
Comment (gavofyork) [inline polkadot/cli/src/panic_hook.rs]: report version & commit hash?

Comment (arkpar) [reply polkadot/cli/src/panic_hook.rs]: next PR

**Pull request merged**